indicators/NO_FUNC_001-004/R/NO_FUNC_001-004.qmd

```diff
@@ -0,0 +1,3139 @@
+unique(ANO.geo$groeftingsintensitet)
+ANO.geo$groeftingsintensitet <- gsub("X", "NA", ANO.geo$groeftingsintensitet)
+unique(ANO.geo$groeftingsintensitet)
+ANO.geo$groeftingsintensitet <- as.numeric(ANO.geo$groeftingsintensitet)
+unique(ANO.geo$groeftingsintensitet)
+
+ANO.geo$bruksintensitet <- gsub("7JB-BA_", "", ANO.geo$bruksintensitet) 
+unique(ANO.geo$bruksintensitet)
+ANO.geo$bruksintensitet <- gsub("X", "NA", ANO.geo$bruksintensitet)
+unique(ANO.geo$bruksintensitet)
+ANO.geo$bruksintensitet <- as.numeric(ANO.geo$bruksintensitet)
+unique(ANO.geo$bruksintensitet)
+
+ANO.geo$beitetrykk <- gsub("7JB-BT_", "", ANO.geo$beitetrykk) 
+unique(ANO.geo$beitetrykk)
+ANO.geo$beitetrykk <- gsub("X", "NA", ANO.geo$beitetrykk)
+unique(ANO.geo$beitetrykk)
+ANO.geo$beitetrykk <- as.numeric(ANO.geo$beitetrykk)
+unique(ANO.geo$beitetrykk)
+
+ANO.geo$slatteintensitet <- gsub("7JB-SI_", "", ANO.geo$slatteintensitet) 
+unique(ANO.geo$slatteintensitet)
+ANO.geo$slatteintensitet <- gsub("X", "NA", ANO.geo$slatteintensitet)
+unique(ANO.geo$slatteintensitet)
+ANO.geo$slatteintensitet <- as.numeric(ANO.geo$slatteintensitet)
+unique(ANO.geo$slatteintensitet)
+
+ANO.geo$tungekjoretoy <- gsub("7TK_", "", ANO.geo$tungekjoretoy) 
+unique(ANO.geo$tungekjoretoy)
+ANO.geo$tungekjoretoy <- gsub("X", "NA", ANO.geo$tungekjoretoy)
+unique(ANO.geo$tungekjoretoy)
+ANO.geo$tungekjoretoy <- as.numeric(ANO.geo$tungekjoretoy)
+unique(ANO.geo$tungekjoretoy)
+
+ANO.geo$slitasje <- gsub("7SE_", "", ANO.geo$slitasje) 
+unique(ANO.geo$slitasje)
+ANO.geo$slitasje <- gsub("X", "NA", ANO.geo$slitasje)
+unique(ANO.geo$slitasje)
+ANO.geo$slitasje <- as.numeric(ANO.geo$slitasje)
+unique(ANO.geo$slitasje)
+
+## check that every point is present only once
+#length(levels(as.factor(ANO.geo$ano_flate_id)))
+#length(levels(as.factor(ANO.geo$ano_punkt_id)))
+summary(as.factor(ANO.geo$ano_punkt_id))
+# there's many double presences, probably some wrong registrations of point numbers,
+
+# fix species names
+ANO.sp$Species <- ANO.sp$art_navn
+unique(as.factor(ANO.sp$Species))
+ANO.sp[,'Species'] <- word(ANO.sp[,'Species'], 1,2) # lose subspecies
+ANO.sp$Species <- str_to_title(ANO.sp$Species) # make first letter capital
+ANO.sp$Species <- gsub("( .*)","\\L\\1",ANO.sp$Species,perl=TRUE) # make capital letters after hyphen to lowercase
+ANO.sp$Species <- gsub("( .*)","\\L\\1",ANO.sp$Species,perl=TRUE) # make capital letters after space to lowercase
+
+## merge species data with indicators
+ANO.sp.ind <- merge(x=ANO.sp[,c("Species", "art_dekning", "parentglobalid")], 
+                y= ind.dat[,c("species", "CC", "SS", "RR", "Light", "Moisture", "Soil_reaction_pH", "Nitrogen","Heat_requirement", "Soil_disturbance","Phosphorus","Grazing_mowing")],
+                by.x="Species", by.y="species", all.x=T)
+summary(ANO.sp.ind)
+
+
+## checking which species didn't find a match
+unique(ANO.sp.ind[is.na(ANO.sp.ind$Moisture),'Species'])
+
+# fix species name issues
+ind.dat <- ind.dat %>% 
```

Comment:
For the future I suggest to make a dictionary for such changes. A simple table with 2 columns old_name and new_name and then use mutate(species = if_else species == "old_name", "new_name", species)).
That would save a lot of code here.

Reply:
Yes. Alternatively to use case_when(), like

```
mutate(species = case_when(
  species == "Species A" ~ "Species B",
  species == "Species C" ~ "Species D"))
```